Shared/Network/WebService.swift

```diff
@@ -0,0 +1,22 @@
+public protocol WebService {
+    func load<T>(resource: Resource<T>, completion: @escaping (T?, Error?) -> ())
+    func download(url: URL, completion: @escaping (Data?) -> ())
+}
+
+final class WebServiceSession: WebService {
+
+    func load<T>(resource: Resource<T>, completion: @escaping (T?, Error?) -> ()) {
```

Comment:
I do note have the certain about extensions yet. Sometimes could be worst to read. In this case i prefer to maintain right in class.

https://medium.com/@theobendixson/how-to-avoid-protocol-orientation-obsessed-programming-a77afb472272#.ip751qkui